<?xml version="1.0"?>
<robot name="cougerbot">
	<link name="world"/>
	<link name="base_link">
		<visual>
			<geometry>
				<cylinder length="0.05" radius="0.1"/>
			</geometry>
			<material name="silver">
				<color rgba="0.75 0.75 0.75 1"/>
			</material>
			<origin rpy="0 0 0" xyz="1 0 0.025"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.05" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="1 0 0.025"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="1 0 0.025"/>
			<inertia ixx="0.0027" iyy="0.0027" izz="0.005" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="fixed1" type="fixed">
		<parent link="world"/>
		<child link="base_link"/>
	</joint>
	
<link name="torso1">
		<visual>
		<geometry>
			<cylinder length="0.5" radius="0.05"/>
		</geometry>
		<material name="silver">
			<color rgba="0.75 0.75 0.75 1"/>
		</material>
		<origin rpy="0 0 0" xyz="0 0 0.25"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.5" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.25"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.25"/>
			<inertia ixx="0.02146" iyy="0.02146" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="hip1" type="continuous">
		<axis xyz="0 0 1"/>
		<parent link="base_link"/>
		<child link="torso1"/>
		<origin rpy="0 0 0" xyz="1 0.0 0.05"/>
	</joint>

<link name="upper_arm1">
	<visual>
		<geometry>
			<cylinder length="0.4" radius="0.05"/>
		</geometry>
		<material name="silver"/>
		<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
			<inertia ixx="0.01396" iyy="0.01396" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="shoulder1" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="torso1"/>
		<child link="upper_arm1"/>
		<origin rpy="0 1.5708 0" xyz="0 -0.1 0.45"/>
	</joint>
	<link name="lower_arm1">
		<visual>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<material name="silver"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
			<inertia ixx="0.01396" iyy="0.01396" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="elbow1" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="upper_arm1"/>
		<child link="lower_arm1"/>
		<origin rpy="0 0 0" xyz="0.0 0.1 0.35"/>
	</joint>
	<link name="hand1">
		<visual>
			<geometry>
				<box size="0.05 0.05 0.05"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.05 0.05 0.05"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<inertia ixx="0.00042" iyy="0.00042" izz="0.00042" ixy="0" ixz="0" iyz="0"/>
			</inertial>
	</link>
	<joint name="wrist1" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="lower_arm1"/>
		<child link="hand1"/>
		<origin rpy="0 0 0" xyz="0.0 0.0 0.425"/>
	</joint>
	
	
	<link name="base_link1">
		<visual>
			<geometry>
				<cylinder length="0.05" radius="0.1"/>
			</geometry>
			<material name="silver">
				<color rgba="0.75 0.75 0.75 1"/>
			</material>
			<origin rpy="0 0 0" xyz="0 0 0.025"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.05" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.025"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.025"/>
			<inertia ixx="0.0027" iyy="0.0027" izz="0.005" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="fixed" type="fixed">
		<parent link="world"/>
		<child link="base_link1"/>
	</joint>
	
	<link name="torso">
		<visual>
		<geometry>
			<cylinder length="0.5" radius="0.05"/>
		</geometry>
		<material name="silver">
			<color rgba="0.75 0.75 0.75 1"/>
		</material>
		<origin rpy="0 0 0" xyz="0 0 0.25"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.5" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.25"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.25"/>
			<inertia ixx="0.02146" iyy="0.02146" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="hip" type="continuous">
		<axis xyz="0 0 1"/>
		<parent link="base_link"/>
		<child link="torso"/>
		<origin rpy="0 0 0" xyz="0.0 0.0 0.05"/>
	</joint>
	
	<link name="upper_arm">
	<visual>
		<geometry>
			<cylinder length="0.4" radius="0.05"/>
		</geometry>
		<material name="silver"/>
		<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
			<inertia ixx="0.01396" iyy="0.01396" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="shoulder" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="torso"/>
		<child link="upper_arm"/>
		<origin rpy="0 1.5708 0" xyz="0.0 -0.1 0.45"/>
	</joint>
	
	<link name="lower_arm">
		<visual>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<material name="silver"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</collision>
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
			<inertia ixx="0.01396" iyy="0.01396" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
	</link>
	<joint name="elbow" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="upper_arm"/>
		<child link="lower_arm"/>
		<origin rpy="0 0 0" xyz="0.0 0.1 0.35"/>
	</joint>
	
	
	<link name="hand">
		<visual>
			<geometry>
				<box size="0.05 0.05 0.05"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.05 0.05 0.05"/>
			</geometry>
		</collision>0
		<inertial>
			<mass value="1.0"/>
			<inertia ixx="0.00042" iyy="0.00042" izz="0.00042" ixy="0" ixz="0" iyz="0"/>
			</inertial>
	</link>
	<joint name="wrist" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="lower_arm"/>
		<child link="hand"/>
		<origin rpy="0 0 0" xyz="0.0 0.0 0.425"/>
	</joint>
	
	
	<link name="cube1">
    <inertial>
      <mass value="1" />
      <origin xyz="0 0 0" />
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="-0.5 0 0.025" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
    </collision>
  </link>
  
  
  <joint name="obj1" type="fixed">
    <parent link="world" />
    <child link="cube1" />
    <origin xyz="-0.5 0 0.025" rpy="0 0 0" />
  </joint>
  
  
  
  
  
  
  <link name="cube2">
    <inertial>
      <mass value="1" />
      <origin xyz="0 0 0" />
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="-0.5 0 0.025" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
    </collision>
  </link>
  
  
  <joint name="obj2" type="fixed">
    <parent link="world" />
    <child link="cube2" />
    <origin xyz="0.5 0 0.025" rpy="0 0 0" />
  </joint>
  
  
  
   <link name="cube3">
    <inertial>
      <mass value="1" />
      <origin xyz="0 0 0" />
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="-0.5 0 0.025" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
    </collision>
  </link>
  
  
  <joint name="obj3" type="fixed">
    <parent link="world" />
    <child link="cube3" />
    <origin xyz="1.5 0 0.025" rpy="0 0 0" />
  </joint>
  
  
  
  
  
  
  
  
  
	
	<ros2_control name="GazeboSimSystem" type="system">
		<hardware>
			<plugin>gazebo_ros2_control/GazeboSystem</plugin>
		</hardware>
		<joint name="hip1">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="shoulder1">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="elbow1">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="wrist1">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="hip">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="shoulder">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="elbow">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
		<joint name="wrist">
			<command_interface name="position"/>
			<state_interface name="position">
				<param name="initial_value">0.0</param>
			</state_interface>
			<state_interface name="velocity"/>
			<state_interface name="effort"/>
		</joint>
	</ros2_control>

<gazebo>
		<plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
			<robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
			<parameters>/home/jettysowmith/ros2_course_ws/src/cougerbot/config/controllers.yaml</parameters>
		</plugin>
	</gazebo>
</robot>
